report/report.go

```diff
@@ -40,14 +41,48 @@ func (h *HTMLReportUploader) DeployReports() []error {
 		h.logger.Printf("- %s", report.Name)
 	}
 
-	var errors []error
-	for _, report := range reports {
+	validatedReports, validationErrors := h.validate(reports)
+	if len(validationErrors) != 0 {
+		h.logger.Warnf("Validation errors:\n")
+
+		for _, validationError := range validationErrors {
+			h.logger.Warnf("- %s\n", validationError)
+		}
+	}
+
+	var uploadErrors []error
+	for _, report := range validatedReports {
 		if err := h.uploadReport(report); err != nil {
-			errors = append(errors, err)
+			uploadErrors = append(uploadErrors, err)
 		}
 	}
 
-	return errors
+	return uploadErrors
+}
+
+func (h *HTMLReportUploader) validate(reports []Report) ([]Report, []error) {
+	var validatedReports []Report
+	var validationErrors []error
+
+	for _, report := range reports {
+		valid := false
+
+		for _, asset := range report.Assets {
+			if strings.HasSuffix(asset.Path, "index.html") {
```

Comment:
Would `/my/path/my_report_index.html` pass this? We need exact "index.html".

Reply:
Yeah, it would. Fixed it in the next commit.